File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, k):
    answer = 0
    heapq.heapify(scoville)
    mix = 0
    
    while scoville:
        first = heapq.heappop(scoville)
        if first >= k:
            return answer
        if not scoville:
            break
        second = heapq.heappop(scoville)
        mix = first + second * 2
        heapq.heappush(scoville, mix)
        answer += 1
    
    return -1

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        Queue<Integer> pq = new PriorityQueue<>();
        for(int i=0 ; i<scoville.length ; i++){
            pq.add(scoville[i]);
        }
        
        while(!pq.isEmpty() && pq.peek() < K){
            int num1 = pq.poll();
            if(pq.isEmpty())break;
            answer++;
            int num2 = pq.poll() * 2;
            pq.add(num1 + num2);
        }
        if(pq.isEmpty())answer = -1;
        return answer;
    }
}